tensorflow/compiler/xla/service/gpu/BUILD

```diff
@@ -880,6 +895,36 @@ tf_cc_test(
     ],
 )
 
+cc_library(
+    name = "cudnn_conv_pad_for_integer_convolutions",
+    srcs = ["cudnn_conv_pad_for_integer_convolutions.cc"],
+    hdrs = ["cudnn_conv_pad_for_integer_convolutions.h"],
+    deps = [
+        ":cudnn_conv_pad_features",
+        ":ir_emission_utils",
+        "//tensorflow/compiler/xla:literal_util",
+        "//tensorflow/compiler/xla:util",
+        "//tensorflow/compiler/xla:window_util",
+        "//tensorflow/compiler/xla/service:hlo_casting_utils",
+        "//tensorflow/compiler/xla/service:hlo_pass",
+    ],
+)
+
+tf_cc_test(
+    name = "cudnn_conv_pad_for_integer_convolutions_test",
```

Comment:
Is it worth the cost of adding two file (classes) and one extra pass? How about:
* move all three classes into a single .cc file,
* change file/pass name `pad_for_tensor_cores` to `pad_for_performance`.

Reply:
I want to point out that these two paddings run under different conditions: pad_for_tensor_cores runs only if IsVoltaOrLater, while it doesn't have this condition.  
I created the new utility class, because the padding/slicing code are so handy that it can be reused by more transforms.  
If you think concise code still more preferable in this case, I am happy to take your suggestion.
pad_for_performance may not be a precise name, as pad_for_integer_convolution is not for performance.  How about pad_for_convolution?

Reply:
I have combined the two padding in one new transform called pad for convolution in the follow-up commit.  I keep the refactoring but reduce the number of files and classes.